app/src/main/java/jayp/bge/playboikotlin/zeb/fragments/hurricankotalina/HurricaneKotalinaSubFragment.kt

```diff
@@ -18,12 +20,63 @@ class HurricaneKotalinaSubFragment: Fragment() {
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
-        binding = FragmentHurricaneKotalinaSubBinding.inflate(layoutInflater,container,false)
+        binding = FragmentHurricaneKotalinaSubBinding.inflate(layoutInflater, container, false)
         return binding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         binding.kotalinaText.text = args.kotalinaText
+        setClickListeners()
+        viewModel.viewStateLiveData.observe(viewLifecycleOwner) { observeViewState(it) }
+    }
+
+    private fun setClickListeners() {
+        binding.kotalinaRequest.setOnClickListener {
+            viewModel.webRequestAddFortune("yoyoyooy", 852, "she called me a kot", 9)
+        }
+    }
+
+    private fun observeViewState(viewState: ZebWebRequestViewState) {
+        when (viewState.webResponse) {
+            is ZebFailure -> onWebFailure(viewState.webResponse.errorMessage)
+            ZebLoading -> showLoading(true)
+            is ZebSuccess -> onWebSuccess(viewState.webResponse.data)
+        }
+    }
+
+    private fun onWebSuccess(data: List<EightBallJson>?) {
+        with(binding) {
+            data?.let { nonNullData ->
+                kotalinaResponse.text = data.toString()
+            } ?: kotlin.run {
+                kotalinaResponse.text = "Oh no! Web response was null :("
+            }
+            showLoading(false)
+            kotalinaScrolling.visibility = View.VISIBLE
```

Comment:
```suggestion
            kotalinaScrolling.visibility = VISIBLE
```

type that in and POW POW the keyboard to import it as just VISIBLE rather than View.VISIBLE. Makes the code easier to read